Epam.TestAutomation.Tests/CareersPageTests.cs

```diff
@@ -37,20 +40,130 @@ public void HoverOverCareerMenuTest()
+    {
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.careersButton)).Build().Perform();
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.jobListingsButton)).Click().Build().Perform();
+        BrowserFactory.Browser.ClickElement(By.XPath("//div[@class='selected-params ']"));
+        Thread.Sleep(3000);
+        BrowserFactory.Browser.ClickElement(By.XPath($"//span[contains(text(),'{SelectSkillsModel.SearchSkill}')]"));
+        BrowserFactory.Browser.ClickElement(careersPage.joinOurTeamFindButton);
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.jobListingsSearchResults)).Build().Perform();
+        Thread.Sleep(3000);
+        List<IWebElement> listOfSearchResults = BrowserFactory.Browser.FindElements(careersPage.jobListingsSearchResultCell).ToList();
+        List<string> resultsToLower = listOfSearchResults.Select(result => result.Text.ToLower()).ToList();
+        var expectedResult = SelectSkillsModel.SearchResult;
+        var Output = string.Join(",", resultsToLower);
+        Assert.That(resultsToLower.All(result => result.Contains(expectedResult)),Is.True, $"The search results are NOT related to your keyword phrase : {Output}");
+    }
+    static List<SelectSkillsModel> GetSearchTestData3()
+    {
+        var json = File.ReadAllText("/Users/alehhramovich/RiderProjects/Epam.TestAutomation/Epam.TestAutomation.Tests.DDT/TestData/DDTSelectedSkills.json");
+        return JsonParser.DeserializeJsonToObjects<SelectSkillsModel>(json);
+    }
+    
+    [Test]
+    [TestCaseSource(nameof(GetSearchTestData4))]
+    public void DDTSelectedSkillsAndLocations(SkillsAndLocationsModel skillsAndLocationsModel)
+    {
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.careersButton)).Build().Perform();
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.jobListingsButton)).Click().Build().Perform();
+        BrowserFactory.Browser.ClickElement(careersPage.keyWord);
+        BrowserFactory.Browser.EnterText(careersPage.keyWord, SkillsAndLocationsModel.SearchPhrase);
+        BrowserFactory.Browser.ClickElement(By.XPath("//span[@class='select2-selection__rendered']"));
+        BrowserFactory.Browser.ClickElement(By.XPath($"//strong[contains(text(),'{SkillsAndLocationsModel.SearchCountry}')]"));
+        BrowserFactory.Browser.ClickElement(By.XPath($"//li[contains(text(),'{SkillsAndLocationsModel.SearchCity}')]"));
+        BrowserFactory.Browser.ClickElement(By.XPath("//div[@class='selected-params ']"));
+        Thread.Sleep(3000);
+        BrowserFactory.Browser.ClickElement(By.XPath($"//span[contains(text(),'{SkillsAndLocationsModel.SearchSkill}')]"));
+        BrowserFactory.Browser.ClickElement(careersPage.joinOurTeamFindButton);
+        action.MoveToElement(BrowserFactory.Browser.FindElement(careersPage.jobListingsSearchResults)).Build().Perform();
+        Thread.Sleep(3000);
```

Comment:
Use the explicit wait instead of Thread.Sleep. And remove all unnecessary actions from the tests (ClickElement and MoveToElement).